src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_NuGet.cs

```diff
@@ -0,0 +1,241 @@
+                packageServiceMock.Setup(s => s.FindPackagesWithType(
+                    NugetOrgSource, "NuGetType", 0, It.IsAny<CancellationToken>()))
+                    .Returns(CreateSearchResult("NuGetPackage", "NuGetType", CreateNameParts("NS1", "NS2")));
+
+                await TestMissingAsync(
+@"
+class C
+{
+    [|NuGetType|] n;
+}", fixProviderData: new FixProviderData(installerServiceMock.Object, packageServiceMock.Object));
+            }
+
+            [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddUsing)]
+            public async Task TestOptionsOffered()
+            {
+                // Make a loose mock for the installer service.  We don't care what this test
+                // calls on it.
+                var installerServiceMock = new Mock<IPackageInstallerService>(MockBehavior.Loose);
+                installerServiceMock.SetupGet(i => i.PackageSources).Returns(NugetPackageSources);
+                installerServiceMock.Setup(s => s.GetInstalledVersions("NuGetPackage"))
+                    .Returns(new[] { "1.0", "2.0" });
+
+                var packageServiceMock = new Mock<IPackageSearchService>();
+                packageServiceMock.Setup(s => s.FindPackagesWithType(
+                    NugetOrgSource, "NuGetType", 0, It.IsAny<CancellationToken>()))
+                    .Returns(CreateSearchResult("NuGetPackage", "NuGetType", CreateNameParts("NS1", "NS2")));
+
+                var data = new FixProviderData(installerServiceMock.Object, packageServiceMock.Object);
+                await TestSmartTagTextAsync(
+@"
+class C
+{
+    [|NuGetType|] n;
+}",
+"Use local version '1.0'",
+index: 0,
+fixProviderData: data);
+
+                await TestSmartTagTextAsync(
+@"
+class C
+{
+    [|NuGetType|] n;
+}",
+"Use local version '2.0'",
+index: 1,
+fixProviderData: data);
+
+                await TestSmartTagTextAsync(
+@"
+class C
+{
+    [|NuGetType|] n;
+}",
+"Find and install latest version",
+index: 2,
+fixProviderData: data);
+            }
+
+            [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddUsing)]
+            public async Task TestInstallGetsCalledNoVersion()
+            {
+                var installerServiceMock = new Mock<IPackageInstallerService>(MockBehavior.Loose);
+                installerServiceMock.SetupGet(i => i.PackageSources).Returns(NugetPackageSources);
+                installerServiceMock.Setup(s => s.TryInstallPackage(
+                    It.IsAny<Workspace>(), It.IsAny<DocumentId>(), "NuGetPackage", /*versionOpt*/ null, It.IsAny<CancellationToken>()));
+
+                var packageServiceMock = new Mock<IPackageSearchService>();
+                packageServiceMock.Setup(s => s.FindPackagesWithType(
+                    NugetOrgSource, "NuGetType", 0, It.IsAny<CancellationToken>()))
+                    .Returns(CreateSearchResult("NuGetPackage", "NuGetType", CreateNameParts("NuGetNamespace")));
```

Comment:
You could extract a method like `CreatePackageService()` here to save the duplication between all these tests.